include/broker/zeek.hh

```diff
@@ -1,11 +1,24 @@
 #pragma once
 
+#include <cstddef>
+#include <iterator>
+
 #include "broker/data.hh"
+#include "broker/detail/assert.hh"
 
 namespace broker::zeek {
 
 const count ProtocolVersion = 1;
 
+/// Metadata attached to Zeek events.
+enum class Metadata : uint8_t {
+  NETWORK_TIMESTAMP = 1,
+  // 1 - 199 is reserved for Zeek. Otherwise free for external
+  // users to experiment with before potentially including
+  // in the reserved range.
+  USER_METADATA_START = 200,
+};
```

Comment:
@ckreibich / @timwoj / @J-Gras   - for extra metadata attached to Zeek events, this is the convention @Neverlord / me came up with for the time being. There might be a handful things coming up or none at all. i don't expect hundreds.

On the wire metadata is actually identified by a `count`, so limiting at 255 is pretty arbitrary currently. There were thoughts of using strings instead of count for a bit of space saving for short identifiers (count takes 8 bytes), but also coming up with with a naming scheme within the metadata "zeek::event::network_time" etc, or "company::product::function::arg1", but I'd forgo this all and keep static and a bit magic numbers (we're coming from a proposal of placing a magic value at a positional place to a bit more extensibility).

IMO the 8bytes on the wire are a bit unfortunate, but it doesn't seem like the end of the world and maybe in the future broker could learn how to more efficiently encode small counts, too, if that's still possible.

Other thoughts around this?

Reply:
Thank you for tackling that swiftly! I am good with everything that supports adding timestamps 😉  Out of curiosity: As I understand, one goal for the extensible event metadata is to allow adding new metadata without the need to touch broker. So I guess this implementation was done under the assumption that `zeek.hh` can and will be moved into Zeek, correct?

Reply:
Yeah, we need one place to define well-known metadata and zeek.hh seems fitting (Dominik has emphasized a few times it's wrongly placed in broker anyhow). The network timestamp will be a common use-case, so also seems fine to have convenience constructor/accessors on Event just for that.

With the proposal here it's possible to add additional metadata to an Event without modifying broker: Even if it's not defined in the `Metadata` enum, can write a Python client pushing `123: "my-metadata"` with events and if we had a generic Zeek side accessor bifs consume it there or consume it with another Python/Go client.

Reply:
> wrongly placed in broker anyhow

I wouldn't say "wrongly". I'd say for "historic reasons" and because of the Python bindings. :)

If we drop the Python bindings, I think this header can move to Zeek.